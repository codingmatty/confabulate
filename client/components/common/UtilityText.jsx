import styled from 'styled-components';

export default styled.div`
  color: ${({ theme }) => theme.color.neutrals[5]};
  font-size: ${({ small }) => (small ? `0.75rem` : `1rem`)};
  margin: 0 0 1em;
  text-transform: uppercase;
`;
