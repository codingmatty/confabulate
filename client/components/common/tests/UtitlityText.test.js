import { cleanup } from 'react-testing-library';
import renderWithTheme from '../../../test-utils/render-with-theme';
import UtilityText from '../UtilityText';

describe('UtilityText', () => {
  afterEach(cleanup);

  it('matches default snapshot', () => {
    const { container } = renderWithTheme(
      <UtilityText>Utility Text</UtilityText>
    );
    expect(container).toMatchSnapshot();
  });

  it('applies small styles', () => {
    const { container } = renderWithTheme(
      <UtilityText small>Utility Text</UtilityText>
    );
    expect(container).toMatchSnapshot();
  });
});
