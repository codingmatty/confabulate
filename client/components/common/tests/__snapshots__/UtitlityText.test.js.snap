// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilityText applies small styles 1`] = `
.c0 {
  color: #7E7E7E;
  font-size: 0.75rem;
  margin: 0 0 1em;
  text-transform: uppercase;
}

<div>
  <div
    class="c0"
  >
    Utility Text
  </div>
</div>
`;

exports[`UtilityText matches default snapshot 1`] = `
.c0 {
  color: #7E7E7E;
  font-size: 1rem;
  margin: 0 0 1em;
  text-transform: uppercase;
}

<div>
  <div
    class="c0"
  >
    Utility Text
  </div>
</div>
`;
